tests/components/neato/test_init.py

```diff
@@ -68,3 +94,24 @@ def mock_controller_login():
     assert entries[0].data[CONF_USERNAME] == USERNAME
     assert entries[0].data[CONF_PASSWORD] == PASSWORD
     assert entries[0].data[CONF_VENDOR] == VENDOR_NEATO
+
+
+async def test_config_entries_not_in_sync_error(hass):
+    """The config entry and configuration.yaml are not in sync, the new configuration is wrong."""
+    MockConfigEntry(domain=NEATO_DOMAIN, data=VALID_CONFIG).add_to_hass(hass)
+
+    assert hass.config_entries.async_entries(NEATO_DOMAIN)
+    with patch(
+        "homeassistant.components.neato.config_flow.NeatoConfigFlow.try_login",
```

Comment:
Generally it's better to patch and mock in the perifery, ie not our code but the library code. That way, if we refactor, we don't need to update tests.

Can we patch the library instead?

Reply:
I think that should be possible. Is it just `patch(pybotvac.Account)` then?

My intention was not to rely on modules other than the one currently being tested.

Reply:
Done